### ISSUE TITLE
Projectors not working in Multiplayer

### ISSUE THREAD

**Issue by khamseen** [KeenSoftwareHouse/SpaceEngineers#268]
We've been using them on our dedicated server for months without any issues but this last week or two it's gotten very unreliable.

1, If I turn the projector on and upload a blueprint, nobody else can see it except me. They have to then load the same blueprint up on the projector for us both to see it. (Also, if we both load different blueprints, we both see whatever we added but not what the other person added.)
2, None of us can build from a projected blueprint if we've loaded it up on the server any more.
3, If we load the blueprint onto the projector in a local save and then upload that save file to the server, it works and everybody can see and build from the blueprint provided nobody removes the blueprint. (Turning the projector on and off does not cause any problems here either.)
4, Projectors no longer remember what they were projecting if the blueprint was loaded on the server, but they do if the blueprint was loaded in single player and then uploaded to the server.

This problem only showed up about two weeks or so ago, three max, as we'd been working on a new station in creative and when we went to work on it in survival we noticed the issue. It's now happened with several of our other ships which we've tried to either build from scratch or just overlay to repair damaged sections.

This is quite an annoying issue given how much we rely on projectors for ship/station builds and repairing ships.

For the record, I tried disabling all mods to make sure it wasn't a conflict. The problem persists even with total vanilla setup.

**Comment by thelastWallE** [comment]
Can the other player see what you have set for the projector in the menu?

**Comment by khamseen** [comment]
You mean which blueprint is loaded? If so then no. When he was looking at the projector in the control panel it said it didn't have a blueprint loaded. But if I load it in local and then upload it to the server, he sees it fine.

I've done a lot of testing with this today to try and resolve it but it simply doesn't have an issue on local. It's solely when trying to load a blueprint on the dedicated server.

**Comment by khamseen** [comment]
Oh, one other thing which may or may not be of relevance. If the projector is on (in dedicated server) but doesn't have any blueprint loaded. When I first go to add a blueprint I then need to toggle it off and on again for it to actually display. Whereas if I load it up locally and add a blueprint, it instantly appears.

This behaviour seems to have coincided with this bug as previously, when loading blueprints on the server it acted the same as in local and instantly appeared.

**Comment by khamseen** [comment]
Ok, did some more testing last night once my brother was available and it actually looks like this issue is limited to certain blueprints. Not sure why as these blueprints have worked ok before. Perhaps some of them have become corrupted or something. 

I'll close this just now as it doesn't appear to be a general bug but more a very bizarre limited event that I'll need to do some more work on to track down.

**Comment by thelastWallE** [comment]
Hi again. Is this your thread on the forum? http://forums.keenswh.com/threads/unusual-issue-with-projectors-and-certain-blueprints.7361824/
If so then you have found the solution to the bug?
If the projector on the grid that is to be blueprinted would be empty then there would be no extra data in the blueprint file i think?
You can reopen the issue if you want.
Maybe they fixed it internal today. But i think not.

**Comment by khamseen** [comment]
Yeah that is my thread. I've pretty much tracked it down to a size issue with blueprints and DS. The projector storing the projection of the blueprint causes an exponential increase in the size of blueprints. I'll reopen this just now and simply link to the relevant bug report thread on the forums where I've put in my over all findings for reference. 

http://forum.keenswh.com/threads/blueprints-of-ships-with-projectors-which-have-active-blueprints-very-bad-things.7361872/
